.github/workflows/check_issue_label.yml

```
@@ -0,0 +1,115 @@
+            // SECURE INPUT HANDLING
+            // ======================
+            const sanitizedLabel = (context.payload.label?.name || '').trim();
+            const actorUsername = (context.actor || '').trim().toLowerCase();
+            const issueNumber = Number(context.issue.number) || 0;
+            const { owner, repo } = context.repo;
+            // Validation checks
+            if (!/^[\w\s-:]+$/i.test(sanitizedLabel)) {
+              throw new Error(`Invalid label format: ${sanitizedLabel}`);
+            }
+            if (!/^[a-z\d-]+$/.test(actorUsername)) {
+              throw new Error(`Invalid username: ${actorUsername}`);
+            }
+            if (!Number.isSafeInteger(issueNumber) || issueNumber < 1) {
+              throw new Error(`Invalid issue number: ${issueNumber}`);
+            }
+            // ======================
+            // TEAM MEMBERSHIP CHECK
+            // ======================
+            async function getTeamMembers() {
+              try {
+                const result = await octokit.graphql({
+                  query: `query ($org: String!, $teamSlug: String!) {
+                    organization(login: $org) {
+                      team(slug: $teamSlug) {
+                        members(first: 100) {
+                          nodes { login }
+                        }
+                      }
+                    }
+                  }`,
+                  org: 'oppia',
+                  teamSlug: 'oppia-good-first-issue-labelers'
+                });
+                return new Set(
+                  result.organization.team.members.nodes
+                    .map(node => node.login.toLowerCase())
+                );
+              } catch (error) {
+                console.error('Team membership check failed:', error);
+                return new Set(); // Fail closed
+              }
+            }
+            // ======================
+            // LABEL ENFORCEMENT
+            // ======================
+            async function enforceLabelRules() {
+              const authorizedMembers = await getTeamMembers();
+              const prSpecificLabels = new Set(['dependencies', 'stale']);
+              // Handle 'good first issue' label
+              if (sanitizedLabel.toLowerCase() === 'good first issue') {
+                if (!authorizedMembers.has(actorUsername)) {
+                  await octokit.rest.issues.removeLabel({
+                    owner,
+                    repo,
+                    issue_number: issueNumber,
+                    name: sanitizedLabel
+                  });
+                  await octokit.rest.issues.createComment({
+                    owner,
+                    repo,
+                    issue_number: issueNumber,
+                    body: `@${actorUsername}, only [oppia-good-first-issue-labelers team members](https://github.com/orgs/oppia/teams/oppia-good-first-issue-labelers) can apply this label.`
```

```suggestion
                    body: `@${actorUsername}, only [oppia-good-first-issue-labelers team members](https://github.com/orgs/oppia/teams/oppia-good-first-issue-labelers) can apply the good first issue label.`
```

if a user adds multiple labels at once, let's make clear which label was problematic